sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -206,7 +217,7 @@ class NodeHttpClient implements HttpClient {
     request: PipelineRequest,
     abortController: AbortController,
     body?: RequestBodyType,
-  ): Promise<http.IncomingMessage> {
+  ): { promise: Promise<http.IncomingMessage>; cleanup: () => void } {
```

Comment:
instead of making these changes to `makeRequest` we can pass the signal directly in the `http.RequestOptions` as this is supported since node16:

https://nodejs.org/docs/v22.14.0/api/http.html#httprequestoptions-callback

The only other modification we'll need to do is have the "error" listener handle checking if it's an `AbortError` and rethrowing similar to what we do in `streamToText`

Reply:
Reverted